crates/core/src/metadata_store.rs

```diff
@@ -44,36 +68,70 @@ pub enum WriteError {
     FailedPrecondition(String),
     #[error("network error: {0}")]
     Network(GenericError),
-    #[error("internal error: {0}")]
-    Internal(String),
+    #[error("other error: {0}")]
+    Other(BoxedMaybeRetryableError),
```

Comment:
Should Network variant be merged into this or is there value in making that distinction to the caller? Feel free to leave it as is if it creates an unnecessary hassle to refactor.